pkg/proxy/integrations/mysqlparser/mysqlparser.go

```diff
@@ -568,19 +568,33 @@ func decodeOutgoingMySQL(requestBuffer []byte, clientConn, destConn net.Conn, h
 	doHandshakeAgain := true
 	prevRequest := ""
 	var requestBuffers [][]byte
+
 	for {
 		configMocks := h.GetConfigMocks()
 		tcsMocks := h.GetTcsMocks()
+		firstSQLMock := func(configMocks []*models.Mock) (*models.Mock, bool) {
```

Comment:
Function can be defined outside and better be named as `GetFirstMysqlMock` and I guess we should also compare the mock with some properties, instead of impetuously returning the first mock. Reason behind can be the case where fist mock written in the yaml itself is not the initial handshake mock .